File: the-xray-tracer/typescript/lib/the-http-flow-stack.ts
import * as cdk from '@aws-cdk/core';
import lambda = require('@aws-cdk/aws-lambda');
import sns_sub = require('@aws-cdk/aws-sns-subscriptions');
import sns = require('@aws-cdk/aws-sns');

export interface HttpFlowStackProps extends cdk.StackProps{
    readonly snsTopicARN: string;
}

export class TheHttpFlowStack extends cdk.Stack {

    httpLambda: lambda.Function;
    constructor(scope: cdk.Construct, id: string, props: HttpFlowStackProps) {
        super(scope, id, props);

        // defines an AWS Lambda resource
        this.httpLambda = new lambda.Function(this, 'httpLambdaHandler', {
            runtime: lambda.Runtime.NODEJS_12_X,
            code: lambda.Code.fromAsset('lambda-fns'),
            handler: 'http.handler',
            tracing: lambda.Tracing.ACTIVE
        });

        let topic = sns.Topic.fromTopicArn(this, 'SNSTopic', props.snsTopicARN);
        topic.addSubscription(new sns_sub.LambdaSubscription(this.httpLambda));
    }
}

File: the-xray-tracer/python/the_xray_tracer/the_http_flow_stack.py
from aws_cdk import (
    aws_lambda as _lambda,
    aws_sns as sns,
    aws_sns_subscriptions as subscriptions,
    core
)


class TheHttpFlowStack(core.Stack):
    def __init__(self, scope: core.Construct, id: str, sns_topic_arn: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        http_lambda = _lambda.Function(self, "httpLambdaHandler",
                                       runtime=_lambda.Runtime.NODEJS_12_X,
                                       handler="http.handler",
                                       code=_lambda.Code.from_asset("lambda_fns"),
                                       tracing=_lambda.Tracing.ACTIVE
                                       )

        topic = sns.Topic.from_topic_arn(self, 'SNSTopic', sns_topic_arn)
        topic.add_subscription(subscriptions.LambdaSubscription(http_lambda))
